static/js/controllers/inbox.js

```diff
@@ -202,7 +212,11 @@ require('moment/locales');
         });
       };
 
-      $scope.$on('ContactUpdated', function() {
+      $scope.$on('ContactUpdated', function(contact) {
+        if (!contact || contact._deleted) {
+          // allow the ContactCtrl to handle contact deletion
+          return;
+        }
```

Comment:
I will try to prevent timeout.
Is there some provision for two deletes happening in quick succession, and the resulting updates returning in the wrong order?